pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -167,32 +168,132 @@ func checkVersion(instance *servingv1alpha1.KnativeServing) error {
 	return nil
 }
 
-// Check for service mesh memberRoll for maistra
-func verifyServiceMeshMemberRoll(instance *servingv1alpha1.KnativeServing) error {
-	smmr := &maistrav1.ServiceMeshMemberRoll{}
-	checkError := func(err error, instance *servingv1alpha1.KnativeServing) error {
-		msg := fmt.Sprintf("Istio not detected; please install ServiceMesh")
-		instance.Status.MarkDependencyMissing(msg)
-		log.Error(err, msg)
+// createIngressNamespace checks for knative-serving-ingress namespace and creates if doesn't exist
+func createIngressNamespace(instance *servingv1alpha1.KnativeServing) error {
+	ns := &v1.Namespace{}
+	ingressNamespace = instance.GetNamespace() + "-ingress"
+	if err := api.Get(context.TODO(), client.ObjectKey{Name: ingressNamespace}, ns); err != nil {
+		if apierrors.IsNotFound(err) {
+			ns.Name = ingressNamespace
+			if err = api.Create(context.TODO(), ns); err != nil {
+				return err
+			}
+			return nil
+		}
 		return err
 	}
-	if err := api.Get(context.TODO(), client.ObjectKey{Namespace: "istio-system", Name: "default"}, smmr); err != nil {
-		return checkError(err, instance)
+	return nil
+}
+
+func applyServiceMesh(instance *servingv1alpha1.KnativeServing) error {
+	const (
+		smcpPath = "deploy/resources/serviceMesh/smcp.yaml"
+		smmrPath = "deploy/resources/serviceMesh/smmr.yaml"
+	)
+	log.Info("Installing serviceMeshControlPlane")
+	if err := installServiceMeshControlPlane(smcpPath, instance); err != nil {
+		return err
+	}
+	// wait for serviceMeshControlPlane condition to be ready before reconciling knative serving component
+	if ready, err := isServiceMeshControlPlaneReady(); !ready {
```

Comment:
This should probably generate an error if `!ready` and return an error only if `err != nil`?

Reply:
Actually `isServiceMeshControlPlaneReady` function return error only when if its false so i feel instead of generating error we can use the actual error ?